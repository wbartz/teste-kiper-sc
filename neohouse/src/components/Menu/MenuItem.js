import PropTypes from 'prop-types';
import React from 'react';
import './index.scss';

const MenuItem = ({ children, icon, onClick, className }) => {
  return (
    <li className={`mdc-list-item ${className}`}>
      <button
        type="button"
        onClick={onClick}
        className={`mdc-button ${onClick ? 'link' : 'text'} btn-link`}
      >
        {icon && <i className="material-icons">{icon}</i>}
        {children}
      </button>
    </li>
  );
};

MenuItem.propTypes = {
  children: PropTypes.string,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  className: PropTypes.string,
};

MenuItem.defaultProps = {
  children: '',
  onClick: undefined,
  icon: '',
  className: '',
};

export default MenuItem;
