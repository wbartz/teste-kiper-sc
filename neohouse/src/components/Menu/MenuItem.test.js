import { shallow } from 'enzyme';
import React from 'react';
import MenuItem from './MenuItem';

describe('MenuItem', () => {
  let wrapper;
  const props = {
    active: true,
    anchor: true,
    onClick: jest.fn(),
    children: '',
  };

  beforeEach(() => {
    wrapper = shallow(<MenuItem {...props} />);
  });

  it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
