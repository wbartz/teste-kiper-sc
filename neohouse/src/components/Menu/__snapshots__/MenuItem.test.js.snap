// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem render without crashing 1`] = `ShallowWrapper {}`;
